### ISSUE TITLE
Refactor Exercise creation

### ISSUE THREAD

**Issue by cjerdonek** [teorth/QED#29]
This issue is to finish simplifying `Exercise` creation in `main.js`.

The steps I suggest are--

1. Finish the `exerciseData` mapping in `main.js`, which has a structure identical to the `lawsData` mapping: https://github.com/teorth/QED/blob/699d583f3ef8b2f8d288259ce96b60857f30db50/docs/js/main.js#L100-L103

2. Add divs of the following form to `index.html`: `<div class="section" id="section-1">Conjunction introduction</div>`. These section divs should be placed between the appropriate exercise divs inside the exercise-container div, so that the section headers and exercises are in the same order that they currently appear in `main.js`:
https://github.com/teorth/QED/blob/699d583f3ef8b2f8d288259ce96b60857f30db50/docs/index.html#L123-L126

3. Add a new `data-law="<lawShortName>"` attribute for the two Exercise objects that use a prior law when being constructed (namely exercises 18.1 and 18.2(b)):
https://github.com/teorth/QED/blob/699d583f3ef8b2f8d288259ce96b60857f30db50/docs/js/main.js#L331-L335

4. Change `main.js` and the `exerciseList.forEach()` loop to iterate instead over the children of the exercise-container div, similar to how the law-container div is now being iterated over:
https://github.com/teorth/QED/blob/699d583f3ef8b2f8d288259ce96b60857f30db50/docs/js/main.js#L70-L74
 If the child has class "section," call `newSection()`. And if the child has class "exercise," then call `exerciseFromData()` or `new Exercise()`, depending on whether the `data-law` attribute is set.

**Comment by teorth** [comment]
This sounds like a good idea.  Tangentially related, I am thinking of tweaking the logic.js code treating special laws (universalSpecification etc.) slightly, to use the shortname of the law rather than the law itself to detect whether the law needs special treatment.  This way we don't actually need to have a specific variable for the special laws.

**Comment by cjerdonek** [comment]
I was actually going to suggest the same thing. You can have your global
Laws object containing all special laws (lawsByShortName, IIRC?), and then
use dot notation on that global variable to access instead of key lookup
(or individually defined variable):
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_accessors

**Comment by teorth** [comment]
I ended up doing a slightly different refactoring than the one you selected, in which I completed your exerciseData to create the exercises, and then used the newSection command to create sections and load the exercises whose name begins with the section name.  Also the exercise constructor can now take three different type of inputs, a pre-existing law, a [givens, conclusion] pair, or a quadruplet consisting of [givens, conclusion, givensTemplate, conclusionTemplate] where [givensTemplate, conclusionTemplate] is the proxy used for matching purposes.  It's a bit of a kludge so perhaps some further refactoring would be in order, but it seems to work for now.
